src/test/java/com/google/sps/UserLoginTest.java

```diff
@@ -0,0 +1,134 @@
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    userServlet = new UserLoginServlet();
+    userServlet.doGet(request, response);
+
+    String responseString = stringWriter.getBuffer().toString().trim();
+    JsonElement responseJsonElement = new JsonParser().parse(responseString);
+
+    JsonObject responseJsonObject = responseJsonElement.getAsJsonObject();
+    String logInUrl = responseJsonObject.get("LogInUrl").getAsString();
+    String logOutUrl = responseJsonObject.get("LogOutUrl").getAsString();
+
+    Assert.assertFalse(logOutUrl.isEmpty());
+    helperLoggedIn.tearDown();
+  }
+
+
+  @Test
+  public void LogOutUrlContainsLogOut() throws ServletException, IOException  {
+
+    helperLoggedIn.setUp();
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    userServlet = new UserLoginServlet();
+    userServlet.doGet(request, response);
+
+    String responseString = stringWriter.getBuffer().toString().trim();
+    JsonElement responseJsonElement = new JsonParser().parse(responseString);
+         
+    JsonObject responseJsonObject = responseJsonElement.getAsJsonObject();
+    String logInUrl = responseJsonObject.get("LogInUrl").getAsString();
+    String logOutUrl = responseJsonObject.get("LogOutUrl").getAsString();
+    
+    Assert.assertTrue(logOutUrl.contains("logout"));
+    helperLoggedIn.tearDown();
+  }
+
+  @Test
+  public void loggedOutUserReturnsLogInUrl() throws ServletException, IOException  {
+    helperLoggedOut.setUp();
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    userServlet = new UserLoginServlet();
+    userServlet.doGet(request, response);
+
+    String response = stringWriter.getBuffer().toString().trim();
+    JsonElement responseJsonElement = new JsonParser().parse(response);
+         
+    JsonObject responseJsonObject = responseJsonElement.getAsJsonObject();
+    String logInUrl = responseJsonObject.get("LogInUrl").getAsString();
+    String logOutUrl = responseJsonObject.get("LogOutUrl").getAsString();
+    
+    Assert.assertFalse(logInUrl.isEmpty()) ;
+    helperLoggedOut.tearDown();
+  }
+   @Test
+  public void logInUrlContainsLogin() throws ServletException, IOException  {
+    helperLoggedOut.setUp();
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+
+    userServlet = new UserLoginServlet();
+    userServlet.doGet(request, response);
+
+    String response = stringWriter.getBuffer().toString().trim();
+    JsonElement responseJsonElement = new JsonParser().parse(response);
+
+    JsonObject responseJsonObject = responseJsonElement.getAsJsonObject();
```

Comment:
All four of your tests use the same setup (lines 115-127 in this test). I think you could make this into a helper method that returns the JsonObject to reduce code being repeated in all the tests.